proxy/envoy/config.go

```diff
@@ -742,15 +747,14 @@ func buildInboundListeners(mesh *proxyconfig.MeshConfig, sidecar proxy.Node,
 				listener.Filters = append([]*NetworkFilter{filter}, listener.Filters...)
 			}
 
-			listeners = append(listeners, listener)
-
 		default:
 			glog.V(4).Infof("Unsupported inbound protocol %v for port %#v", protocol, servicePort)
 		}
-	}
 
-	for _, listener := range listeners {
-		applyInboundAuth(listener, mesh)
+		if listener != nil {
+			mayApplyInboundAuth(listener, mesh, endpoint.ServicePort.AuthenticationPolicy)
+			listeners = append(listeners, listener)
+		}
```

Comment:
Can you validate that this works for headless services? In our tests, we are disabling auth for headless services because it doesn't work. If you could fix that as well, while you are here, it would be great.

Reply:
Not quite sure what does it involve :). May be different PR?

Reply:
+1 that needs to be fixed - but smaller PRs are easier to review and test.